src/confu/schema/core.py

```diff
@@ -3,12 +3,18 @@
 
 These can be imported directly from `confu.schema`
 """
+from __future__ import annotations
 
 import collections.abc
 import configparser
 import copy
 import os
 from inspect import isclass
+from typing import Any, Callable
+from typing import Dict as typing_Dict
```

Comment:
i think in these cases where typing clashes with confu schema types (confu.Dict vs typing.Dict) we should probably handle it via
```
import typing
```

and then use `typing.Dict` for the type hints

@jason20c